In [1]:
import pandas as pd
ce=pd.read_csv("/content/Cancer.csv")
print(ce.head())
print(ce.tail())

  Country or Territory  \
0          Afghanistan   
1              Algeria   
2           Azerbaijan   
3              Albania   
4              Armenia   

  Cancer deaths attributable to alcohol\nProportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016  \
0                                                0.5                                                                                      
1                                                1.1                                                                                      
2                                                3.2                                                                                      
3                                                4.9                                                                                      
4                                                  5                                                                                      

  Smokin

The head and tail of the dataset are the same meaning the dataset has been copied successfully

In [ ]:
#checking the number of rows and columns
ce.shape

(207, 27)

There are 207 rows and 27 columns

In [5]:
#checking if there are any duplicates
ce.duplicated().sum()

3

There are 3 duplicated rows

In [4]:
#we drop the duplicated fields
ce.drop_duplicates()

Country or Territory  \
0            Afghanistan   
1                Algeria   
2             Azerbaijan   
3                Albania   
4                Armenia   
..                   ...   
202              Réunion   
203     French Polynesia   
204           Guadeloupe   
205                 Guam   
206           Martinique   

    Cancer deaths attributable to alcohol\nProportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016  \
0                                                  0.5                                                                                      
1                                                  1.1                                                                                      
2                                                  3.2                                                                                      
3                                                  4.9                                                                                      
4                                                    5                                                                                      
..                                                 ...                                                                                      
202                                            No data                                                                                      
203                                            No data                                                                                      
204                                            No data                                                                                      
205                                            No data                                                                                      
206                                            No data                                                                                      

    Smoking prevalence male\nPrevalence (%) of daily smoking for men  \
0                                                 21.4                 
1                                                 17.5                 
2                                                 36.7                 
3                                                 40.9                 
4                                                 43.5                 
..                                                 ...                 
202                                            No data                 
203                                            No data                 
204                                            No data                 
205                                            No data                 
206                                            No data                 

    Smoking prevalence female\nPrevalence (%) of daily smoking for women  \
0                                                    7                     
1                                                  2.2                     
2                                                  0.3                     
3                                                  6.1                     
4                                                  1.5                     
..                                                 ...                     
202                                            No data                     
203                                            No data                     
204                                            No data                     
205                                            No data                     
206                                            No data                     

    Cancers attributable to infections\nProportion of cancers attributable to infections (%), by country  \
0                                                   16                                  

We have dropped the duplicated data fields as seen by the reduction of rows by 3 from 207 to 204 as there were 3 duplicated rows

In [8]:
#checking the total number of# empty felds
print(ce.isnull().sum().sum())

0


There are no empty fields in the dataset

In [7]:
#checking which datatypes are present in dataset
ce.dtypes

Country or Territory                                                                                                                                          object
Cancer deaths attributable to alcohol\nProportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016                         object
Smoking prevalence male\nPrevalence (%) of daily smoking for men                                                                                              object
Smoking prevalence female\nPrevalence (%) of daily smoking for women                                                                                          object
Cancers attributable to infections\nProportion of cancers attributable to infections (%), by country                                                          object
Obesity prevalence male\nInternational variation in the prevalence of obesity, 2016                                                                           object
Obesity pr

Every colummn's datatype is object


In [ ]:
#summary of the dataset
print("Summary of the dataset ")
print(ce.describe())

Summary of the dataset 
       Country or Territory  \
count                   207   
unique                  204   
top                 Lesotho   
freq                      2   

       Cancer deaths attributable to alcohol\nProportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016  \
count                                                 207                                                                                      
unique                                                 96                                                                                      
top                                               No data                                                                                      
freq                                                   23                                                                                      

       Smoking prevalence male\nPrevalence (%) of daily smoking for men  \
count                   

In [10]:
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# Unique values in categorical columns
print("\nUnique Values in Categorical Columns:")
for col in ce.select_dtypes(include='object').columns:
    print(f"{col}: {ce[col].nunique()} unique values")
    print(ce[col].value_counts())
    print()

# Correlation matrix for numerical columns
print("\nCorrelation Matrix for Numerical Columns:")
corr_matrix = ce.select_dtypes(include=np.number).corr()
print(corr_matrix)

# Correlation heatmap
plt.figure(figsize=(10, 8))
sns.heatmap(corr_matrix, annot=True, cmap='coolwarm', fmt=".2f")
plt.title('Correlation Heatmap')
plt.show()




Unique Values in Categorical Columns:
Country or Territory: 204 unique values
Country or Territory
Lesotho                                  2
Timor-Leste                              2
Bahrain                                  2
Afghanistan                              1
Venezuela, Bolivarian Republic of        1
                                        ..
Democratic People's Republic of Korea    1
Republic of Korea                        1
Kuwait                                   1
Kazakhstan                               1
Martinique                               1
Name: count, Length: 204, dtype: int64

Cancer deaths attributable to alcohol
Proportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016: 96 unique values
Cancer deaths attributable to alcohol\nProportion (%) of cancer deaths caused by alcohol drinking in men ages 15 years or older, 2016
No data    23
6.6         6
1.5         5
7.4         4
8.2         4
           ..
3           1
14.5

ValueError: zero-size array to reduction operation fmin which has no identity

<Figure size 1000x800 with 0 Axes>